File: test/cpp/LIM2Metrics/CohesionMetrics/LCOM5_MethodCall/LCOM5_MethodCall.cpp
//LCOM5 = 1
class SuperBase{
public:
    void superBaseMethod(){}
};
//LCOM5 = 1
class Base : public SuperBase{
public:
    void baseMethod(){}
};

//LCOM5 = 4
class MyClass : public Base{
    
    int localMethod(){}
    
    void foo_localMethod(){
        localMethod();
    }
    
    void goo_localMethod(){
        localMethod();
    }
    
    void foo_baseMethod(){
        baseMethod();
    }
    
    void goo_baseMethod(){
        baseMethod();
    }
    
    void foo_superBaseMethod(){
        superBaseMethod();
    }
    
    void goo_superBaseMethod(){
        superBaseMethod();
    }
    
    void foo_call(){
        goo_call();
    }
    
    void goo_call(){
        foo_call();
    }
};    

int main(){
    return 0;
}

File: test/java/LIM2Metrics/CohesionMetrics/LCOM5_MethodCall/LCOM5_MethodCall.java
package regtest;

//LCOM5 = 1
class SuperBase{
  public void superBaseMethod(){}
}
//LCOM5 = 1
class Base extends SuperBase{
  public void baseMethod(){}
}

//LCOM5 = 5
class LCOM5_MethodCall extends Base{ 
  public void localMethod(){}
    
  public void foo_localMethod(){
    localMethod();
  }
  
  public void goo_localMethod(){
    localMethod();
  }
  
  public void foo_baseMethod(){
    baseMethod();
  }
  
  public void goo_baseMethod(){
    baseMethod();
  }
  
  public void foo_superBaseMethod(){
    superBaseMethod();
  }
  
  public void goo_superBaseMethod(){
    superBaseMethod();
  }
  
  public void foo_call(){
    goo_call();
  }
  
  public void goo_call(){
    foo_call();
  }

  public static void main(String[] args){}
}